### DISCUSSION TITLE
select2 in a nested x-data seems to break

### DISCUSSION THREAD

**Discussion by cbaconnier** [livewire/livewire#1120] [4. General]
I'm not entirely sure why but I can't make this to work.

This is the closest reproducible example I can give you;
```blade
<div x-data="{open: false}">
    <div x-data="{selected: 1}"
         x-init="
            $($refs.select).select2({ placeholder: selected })
           .on('change', function(e) {
                @this.set('category_id', e.target.value);
            });
        "
    >
        <div wire:ignore>
            <select x-ref="select"
                    class="select"
                    name="category_id"
                    autocomplete="off"
            >
                @foreach(['foo', 'bar'] as $value => $option)
                    <option value="{{ $value }}">
                        {{ $option }}
                    </option>
                @endforeach
            </select>
        </div>

    </div>
</div>
```

When changing the value, livewire triggers the request but then I'm getting the javascript error from select2
> Uncaught (in promise) TypeError: can't access property "contains", adown is null

Removing `x-data="{open: false}"` fix the problem. But I need it since my select2 is inside a modal.

**Comment by nuernbergerA** [comment]
Try something like this?

```blade
<div>
    <div x-data="{ open: false }" wire:ignore>
        <button @click="open = true">Open</button>
        <div class="modal" x-show="open" @click.away="open = false">
            <div x-data
                 x-init="
            $($refs.select).select2({
                placeholder: 'Select something'
            }).on('change', function(e) {
                @this.set('category_id', e.target.value)
            });
        ">
                <select
                    x-ref="select"
                    class="select"
                    wire:model="category_id"
                    autocomplete="off"
                >
                    <option></option>
                    @foreach(['foo', 'bar'] as $value => $option)
                        <option value="{{ $value }}">
                            {{ $option }}
                        </option>
                    @endforeach
                </select>
            </div>
        </div>
    </div>

    <pre>
ID: {{ $category_id }}
    </pre>
</div>
```

**Reply by cbaconnier** [reply]
Thanks!
Assuming I only had `category_id` this could work. But I have a complete form in my modal with few others inputs using `wire:model="first_name"` and and so on. As far as I understand `wire:ignore`, those would not work anymore. Am I right?

**Reply by nuernbergerA** [reply]
wire:model work's and is not interfering with
wire:ignore

But it would be easy with the actual code to give you the right advise.

**Reply by cbaconnier** [reply]
I didn't know that! Thanks :)
https://gist.github.com/cbaconnier/b730008301bfdfb00ccc95c8e5b98f7d 

It does work, but every fields are disabled on the modal after submitting the form.
I know there is an issue about [select2 being disabled](https://github.com/livewire/livewire/issues/766). I'm already take care of [here](https://gist.github.com/cbaconnier/b730008301bfdfb00ccc95c8e5b98f7d#file-select-blade-php-L61) but it's still disabled alongside the others inputs

What I did in short; 
I have form to edit a project `edit-project.blade.php`
- The user can search a client or create a new one through the modal.
- There's a button to open the modal using an event

In the modal `create-client-modal.blade.php`
- The user create the client and select a category.
- Then fire an event when created

In `EditProject.php` when `client-created` is received
- The edit project will update his select with the newly client created. _(That's why you see so much weird stuff happening in select.blade.php)_
- Then the user save the project

**Comment by cbaconnier** [comment]
I did a reproducible example:
https://github.com/cbaconnier/livewire-select2/commit/0457ae140201128cf84abbb3ada95cd3e16c1390

```
git clone git@github.com:cbaconnier/livewire-select2.git 
cd livewire-select2
composer install
php artisan serve
```

Since I don't have this issue with pikaday, I'm thinking this is specific to select2. I'm open to any other (multi) select picker that also works with ajax though.

Edit: It seems to work well with [chosen](https://github.com/harvesthq/chosen) even though it's deprecated.

**Reply by nuernbergerA** [reply]
try https://joshuajohnson.co.uk/Choices/

**Reply by cbaconnier** [reply]
Thanks ! It does the job!
